internal/resources/kubernetessecret/resource_secret_mock_test.go

```diff
@@ -0,0 +1,220 @@
+
+	// POST
+	secretSpec := getMockSpec()
+	postRequestModel := &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecret{
+		FullName: &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretFullName{
+			Name:          testConfig.SecretName,
+			OrgID:         OrgID,
+			ClusterName:   testConfig.ClusterName,
+			NamespaceName: testConfig.NamespaceName,
+		},
+		Spec: &secretSpec,
+		Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+			ParentReferences: nil,
+			Description:      "resource with description",
+			Labels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+			UID:             "secret1",
+			ResourceVersion: "v1",
+		},
+	}
+
+	reference := objectmetamodel.VmwareTanzuCoreV1alpha1ObjectReference{
+		Rid: "test_rid",
+		UID: "test_uid",
+	}
+	referenceArray := make([]*objectmetamodel.VmwareTanzuCoreV1alpha1ObjectReference, 0)
+	referenceArray = append(referenceArray, &reference)
+
+	postResponseModel := &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecret{
+		FullName: &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretFullName{
+			Name:          testConfig.SecretName,
+			OrgID:         OrgID,
+			ClusterName:   testConfig.ClusterName,
+			NamespaceName: testConfig.NamespaceName,
+		},
+		Spec: &secretSpec,
+		Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+			ParentReferences: nil,
+			Description:      "resource with description",
+			Labels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+			UID:             "secret1",
+			ResourceVersion: "v1",
+		},
+	}
+
+	postRequest := &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretRequest{
+		Secret: postRequestModel,
+	}
+
+	postResponse := &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretResponse{
+		Secret: postResponseModel,
+	}
+
+	// GET Network Secret mock setup
+	getModel := &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecret{
+		FullName: &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretFullName{
+			Name:          testConfig.SecretName,
+			OrgID:         OrgID,
+			ClusterName:   testConfig.ClusterName,
+			NamespaceName: testConfig.NamespaceName,
+		},
+		Spec: &secretSpec,
+		Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+			ParentReferences: nil,
+			Description:      "resource with description",
+			Labels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+			UID:             "secret1",
+			ResourceVersion: "v1",
+		},
+		Status: &status.VmwareTanzuManageV1alpha1ClusterNamespaceStatus{
+			Conditions: map[string]status.VmwareTanzuCoreV1alpha1StatusCondition{
+				"Ready": {
+					Reason: "made successfully",
+				},
+			},
+		},
+	}
+
+	getResponse := &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceGetSecretResponse{
+		Secret: getModel,
+	}
+
+	postEndpoint := fmt.Sprintf("https://%s/v1alpha1/clusters/%s/namespaces/%s/secrets", endpoint, testConfig.ClusterName, testConfig.NamespaceName)
```

Comment:
Can you use the existing helper method to construct the URL as we do it in the client files?

Reply:
updated PTAL